Tribler/Policies/BoostingSource.py

```diff
@@ -0,0 +1,639 @@
+                    tdef.save(torrent_fname)
+                except ValueError, err:
+                    self._logger.error("Could not parse/save torrent, skipping %s. Reason: %s",
+                                       item_torrent['url'], err.message +
+                                       ", metainfo is "+("not " if metainfo else "")+"None")
+                except IOError, ioerr:
+                    # can't save to disk. Ignore this swarm
+                    self._logger.error("IO error, check %s. Message : %s",
+                                       self.boosting_manager.settings.credit_mining_path, ioerr.message)
+                    return
+                if tdef:
+                    # Create a torrent dict.
+                    torrent_values = [item_torrent['title'], tdef, tdef.get_creation_date(), tdef.get_length(),
+                                      len(tdef.get_files()), -1, -1, self.enabled, {}]
+                    self.torrents[sha1(item_torrent['url']).digest()] = dict(zip(torrent_keys, torrent_values))
+
+                    try:
+                        # manually generate an ID and put this into DB
+                        self.session.lm.torrent_db.addOrGetTorrentID(sha1(item_torrent['url']).digest())
+                        self.session.lm.torrent_db.addExternalTorrent(tdef)
+
+                        # create Torrent object and store it
+                        self.torrent_mgr.load_torrent(Torrent.fromTorrentDef(tdef))
+                    except AttributeError, err:
+                        # if we can't find torrent_db, fallback
+                        self._logger.error("Can't do %s. Return gracely as failed.", err.message)
+                        return
+                    # self.gui_util.torrentsearch_manager.loadTorrent(Torrent.fromTorrentDef(tdef))
+
+                    # Notify the BoostingManager and provide the real infohash.
+                    if self.callback:
+                        self.callback(self.source, tdef.get_infohash(), self.torrents[sha1(
+                            item_torrent['url']).digest()])
+
+            def __success_cb(response, item_dict, torrent_filename):
+                defer = readBody(response)
+                defer.addCallback(__cb_body, item_dict, torrent_filename)
+                return defer
+
+            regex_unescape_xml = re.compile(r"\&\#(x?[0-9a-fA-F]+);")
+
+            for item in feed_status['items']:
+                # Not all RSS feeds provide us with the infohash,
+                # so we use a fake infohash based on the URL to identify the torrents.
+                url = regex_unescape_xml.sub(ent2chr, item['url'])
+                infohash = sha1(url).digest()
+                if infohash not in self.torrents:
+                    # Store the torrents as rss-infohash_as_hex.torrent.
+                    torrent_filename = os.path.join(self.boosting_manager.settings.credit_mining_path,
```

Comment:
Why are we storing this torrent on disk? Maybe we can store it in the LevelDB store.